### PR TITLE
En 2759 rewards shard processor 2

### PR DESCRIPTION
**Pull request by AdoAdoAdo**
Fixes on verification of other chain block signature
Fixes for fee transactions

Still required:
1. integration test for verification block signatures of other chains - will be done in separate pull request after merging integration tests refactorization from development branch  
2. for activating fee transactions - separate miniblock of transaction fees and preprocessor - will be done in separate pull request

unit tests              - passed
integration tests  - passed
system tests        - passed (without transaction fees)

### CODE REVIEW

File: sharding/indexHashedNodesCoordinator.go [hunk 1]
```diff
@@ -152,16 +163,20 @@ func (ihgs *indexHashedNodesCoordinator) GetValidatorsPublicKeys(randomness []by
...
-	shardEligibleLen := uint16(len(ihgs.nodesMap[ihgs.shardId]))
+	shardEligibleLen := uint16(len(ihgs.nodesMap[shardId]))
 	invalidSelection := selectionLen < shardEligibleLen
 
 	if invalidSelection {
 		return nil, ErrEligibleSelectionMismatch
 	}
 
-	cSize := ihgs.consensusGroupSize()
+	cSize := ihgs.consensusGroupSize(shardId)
```
Comment (SebastianMarian) [inline sharding/indexHashedNodesCoordinator.go]: Maybe another more explicit name: cnsSize or consensusSize

File: sharding/indexHashedNodesCoordinator.go [hunk 2]
```diff
@@ -3,6 +3,8 @@ package sharding
 import (
 	"bytes"
 	"encoding/binary"
+	"fmt"
```
[the remark this answers is not in the document]
Comment (AdoAdoAdo) [reply sharding/indexHashedNodesCoordinator.go]: done

File: sharding/indexHashedNodesCoordinator.go [hunk 3]
```diff
@@ -80,37 +80,48 @@ func (ihgs *indexHashedNodesCoordinator) SetNodesPerShards(nodes map[uint32][]Va
...
+		return nil, ErrInvalidShardId
+	}
+
 	if ihgs == nil {
 		return nil, ErrNilRandomness
 	}
 
 	tempList := make([]Validator, 0)
-	cSize := ihgs.consensusGroupSize()
+	cSize := ihgs.consensusGroupSize(shardId)
```
[the remark this answers is not in the document]
Comment (AdoAdoAdo) [reply sharding/indexHashedNodesCoordinator.go]: done

File: sharding/indexHashedNodesCoordinator.go (hunk 1 above)
Comment (AdoAdoAdo) [reply sharding/indexHashedNodesCoordinator.go]: done

File: consensus/spos/commonSubround/subroundStartRound.go [hunk 4]
```diff
@@ -181,11 +181,13 @@ func (sr *SubroundStartRound) generateNextConsensusGroup(roundIndex int64) error
 		}
 	}
 
-	randomSource := fmt.Sprintf("%d-%s", roundIndex, core.ToB64(currentHeader.GetRandSeed()))
+	randomSource := currentHeader.GetRandSeed()
```
[the remark this answers is not in the document]
Comment (AdoAdoAdo) [reply consensus/spos/commonSubround/subroundStartRound.go]: done

**Pull request merged**